GSA/digitalgov.gov
# Reduce responsive image variants

## Summary
Updates code to only generate and serve `mobile (800)` and `desktop (1200)` responsive sizes.

## Preview

| pages that display 800 or 1200                                                                                                                                                                                                                                                                                                       |
| ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------ |
| [it-warning-banners-how-gsa-is-working-to-stop-unnecessarily-frightening-users](https://federalist-466b7d92-5da1-4208-974f-d61fd4348571.sites.pages.cloud.gov/preview/gsa/digitalgov.gov/nl-update-responsive-images/2024/08/07/it-warning-banners-how-gsa-is-working-to-stop-unnecessarily-frightening-users/)                      |
| [evaluating-your-agency-accessibility-just-got-a-whole-lot-easier-with-gsa-openacr-editor](https://federalist-466b7d92-5da1-4208-974f-d61fd4348571.sites.pages.cloud.gov/preview/gsa/digitalgov.gov/nl-update-responsive-images/2024/07/08/evaluating-your-agency-accessibility-just-got-a-whole-lot-easier-with-gsa-openacr-editor) |
| [a-successful-vibrant-productive-career-for-federal-employees](https://federalist-466b7d92-5da1-4208-974f-d61fd4348571.sites.pages.cloud.gov/2019/08/05/a-successful-vibrant-productive-career-for-federal-employees)                                                                                                                |
| [making-gsa-public-art-collection-more-accessible](https://federalist-466b7d92-5da1-4208-974f-d61fd4348571.sites.pages.cloud.gov/preview/gsa/digitalgov.gov/nl-update-responsive-images/2023/12/08/making-gsa-public-art-collection-more-accessible/)                                                                                |


> [!NOTE]
> For some pages, there is only one variant, see below

| pages that display other sizes                                                                                                                                                                                                 |
| ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------ |
| [event/2020/01/14/socialgov-winter-meeting/ (only displays 200px)](https://federalist-466b7d92-5da1-4208-974f-d61fd4348571.sites.pages.cloud.gov/preview/gsa/digitalgov.gov/nl-update-responsive-images//event/2020/01/14/socialgov-winter-meeting/) |


## Solution
Simplified responsive variants from 4 to 2 sizes. Serving one size for mobile and desktop will not impact performance and quality.

| viewport            | size |
| ------------------- | ---- |
| mobile/tablet       | 800  |
| desktop/large scree | 1200 |


<details><summary>What you should see in network panel</summary>
<p>

<img width="1194" alt="image (1)" src="https://github.com/user-attachments/assets/9b895212-2f48-45d1-9f25-538a16a932f6">
<img width="1195" alt="image (2)" src="https://github.com/user-attachments/assets/1c5d5a4c-9747-4113-b7fa-74a465c2cfa3">

</p>
</details> 

Updates the `file-process.js` to generate the above two sizes.
Updates `img.html` partial/short code to display above two sizes.

Other variant sizes will be served if the `800` or `1200` is not present.

## How To Test

1. Visit a preview page above
2. Change viewport from desktop to mobile to see only two sizes loaded
3. When an image has no `800` or `1200` size, other image size is served
